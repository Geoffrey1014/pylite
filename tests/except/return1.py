def before_return():
    print 'before return'

def main():
    try:
        print 'try'
        return before_return()
    finally:
        print 'finally' 

print 'global start'
main()
print 'global end'
