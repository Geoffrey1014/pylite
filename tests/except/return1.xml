lambda
  name __init__
  param_count 0
  local_count 0
  label_count 0
  insn_vector
    load_const c0
    store_global c1
    load_const c2
    store_global c3
    load_const c4
    print
    load_global c3
    call 0
    pop
    load_const c5
    print
    load_const c6
    return
  const_vector
    lambda
      name before_return
      param_count 0
      local_count 0
      label_count 0
      insn_vector
        load_const c0
        print
        load_const c1
        return
      const_vector
        "before return"
        None
      guard_vector
      line_number_map 2 2 1 1
    before_return
    lambda
      name main
      param_count 0
      local_count 2
      label_count 4
      insn_vector
        label L0
        load_const c0
        print
        load_global c1
        call 0
        store_local -1
        finally_call L2
        load_local -1
        return
        label L1
        jmp L3
        label L2
        store_local -2
        load_const c2
        print
        finally_exit -2
        label L3
        finally_call L2
        load_const c3
        return
      const_vector
        "try"
        before_return
        "finally"
        None
      guard_vector
        guard
          start L0
          end L1
          handler L2
      line_number_map 6 6 6 7 7 7 7 7 7 6 6 6 6 9 9 6 6 6 4 4
    main
    "global start"
    "global end"
    None
  guard_vector
  line_number_map 1 1 4 4 11 11 12 12 12 13 13 1 1
