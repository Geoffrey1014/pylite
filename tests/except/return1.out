global start
try
before return
finally
global end
